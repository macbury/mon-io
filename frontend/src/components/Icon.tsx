import React from 'react'
import { useTheme } from 'styled-components/native'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'

interface IIconProp {
  name: string;
  size: number;
  color?: string;
}

export default function Icon({ name, color, ...rest } : IIconProp) {
  const [type, key] = name.split(':')
  const { colors } = useTheme()
  const selectedColor = color || colors.icon

  switch(type) {
    case 'MaterialIcons':
      return <MaterialIcon name={key} color={selectedColor} {...rest} />

    case 'MaterialCommunityIcons':
      return <MaterialCommunityIcons name={key} color={selectedColor} {...rest} />

    case 'FontAwesome':
      return <FontAwesomeIcon name={key} color={selectedColor} {...rest} />

    default:
      throw `Could not find type for icon: ${type} and key ${key}`
  }
}