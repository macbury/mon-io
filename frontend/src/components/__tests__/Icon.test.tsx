import React from 'react'
import { shallow } from 'enzyme'
import snapshot from 'enzyme-to-json'
import Icon from '../Icon'

jest.mock('styled-components/native', () => ({
  useTheme: () => ({
    
  })
}))

it('renders correctly', () => {
  const props = {
    name: 'MaterialIcons:repeat',
    size: 32,
    color: '#fff'
  }

  const component = shallow(<Icon {...props}/>)
  expect(snapshot(component)).toMatchSnapshot()
})