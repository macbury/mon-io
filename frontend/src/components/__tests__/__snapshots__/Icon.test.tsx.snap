// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Icon
  allowFontScaling={false}
  color="#fff"
  name="repeat"
  size={32}
/>
`;
